ifttt/watches/gcloud_datastore_aggregation.py

```diff
@@ -19,17 +22,27 @@ def __repr__(self):
         return "AggregatedDatastoreWatch '{}'".format(self.name)
 
     def collect_activations(self):
-        collector = list()
+        collector = collections.defaultdict(list)
 
         query = self.client.query(kind=self.kind)
         for result in query.fetch():
             value = result.get(self.field)
             if not value:
                 continue  # TODO: aggregates across nulls
 
-            collector.append(value)
+            context = None
+            if self.context_field:
+                context = result.get(self.context_field)
+                context = context or None  # TODO: consider handling nulls vs ''
```

Comment:
Could put this on one line with the same behaviour for '' returns:

`context = result.get(self.context_field) or None`

Reply:
True -- just kept it separate to make the TODO a bit more explicit. Theoretically, it might be useful to handle `''` and `None` differently.

Reply:
Okay, in the name of readability it's fine.